unstructured/ingest/connector/discord.py

```diff
@@ -131,7 +131,7 @@ def __init__(
 
     def initialize(self):
         """Verify that can get metadata for an object, validates connections info."""
-        os.mkdir(self.standard_config.download_dir)
+        os.makedirs(self.standard_config.download_dir, exist_ok=True)
```

Comment:
funny caught this in another PR I'm working on, will have a fix for it. I think we actually don't want the folder creation to happen here at all and instead should happen on actually writing the file in the IngestDoc. reason being: if the IngestDoc were say ever processed somewhere (i.e. a different machine) other than where the the Connector was initialized, it could fail because it assumes that folder exists.

Reply:
does it? this looked specific to this discord connector. I haven't looked at all with a close eye, but it seemed that most were following [this convention](https://github.com/Unstructured-IO/unstructured/blob/8104fe274c935dbb824917206b2cacedb71fca99/unstructured/ingest/connector/confluence.py#L119) of making the directory when attempting to write the file (in IngestDoc). which is ideally just what we want to do here has well.